### PR TITLE
Cleanup any mention of --transition flag

### PR DESCRIPTION
**Pull request by gtrepta**
fixes: #1936

### CODE REVIEW

**Comment by Scott-Guest** [comment]
I textually searched the codebase for `transition`, and there are a few lingering references missed by this PR:
- https://github.com/runtimeverification/k/blob/9fa1bf94cc86899b8f50cd47c7b477600149c0f7/k-distribution/pl-tutorial/1_k/4_imp%2B%2B/README.md?plain=1#L17
- https://github.com/runtimeverification/k/blob/9fa1bf94cc86899b8f50cd47c7b477600149c0f7/k-distribution/pl-tutorial/1_k/4_imp%2B%2B/lesson_8/imp.md?plain=1#L75
- https://github.com/runtimeverification/k/blob/9fa1bf94cc86899b8f50cd47c7b477600149c0f7/k-distribution/pl-tutorial/2_languages/1_simple/1_untyped/simple-untyped.md?plain=1#L1072
- https://github.com/runtimeverification/k/blob/9fa1bf94cc86899b8f50cd47c7b477600149c0f7/k-distribution/pl-tutorial/2_languages/1_simple/1_untyped/simple-untyped.md?plain=1#L1164

File: k-distribution/pl-tutorial/1_k/4_imp++/lesson_3/README.md [hunk 1]
```diff
@@ -31,99 +31,6 @@ the `krun` tool with the option `--search`. For example:
```
Comment (Scott-Guest) [inline k-distribution/pl-tutorial/1_k/4_imp++/lesson_3/README.md:31]: We've removed almost everything from this lesson - the title is "Tagging; Transition Kompilation Option", and we no longer explain tagging nor `--transition`. We should consider just deleting it entirely.

The only remaining content is pretty short and could be easily moved into other lessons:
- defining the variable increment rule
- describing `--search`

File: k-distribution/pl-tutorial/1_k/4_imp++/lesson_8/imp.md [hunk 2]
```diff
@@ -563,107 +563,3 @@ This rule acts like a ``tail recursion'' optimization, but for blocks. */
...
-model. Depending upon for what you want to use the generated language model,
-you may need to kompile the definition using various options. We here discuss
-these options.
-
-To tell the **K** tool to exhaustively explore all the behaviors due to the
-non-determinism of addition, division, and threads, we have to kompile
-with the command:
-
-```shell script
-kompile imp.k --transition="addition division lookup assignment increment read print"
```
Comment (Scott-Guest) [inline k-distribution/pl-tutorial/1_k/4_imp++/lesson_8/imp.md:579]: Most of these `addition`, `division`, `lookup`, etc. tags are unneeded now, so we should remove the corresponding `group(_)` attributes on the rules. Similarly in the other lessons too.

File: k-distribution/pl-tutorial/1_k/4_imp++/lesson_3/README.md [hunk 3]
```diff
@@ -31,99 +31,6 @@ the `krun` tool with the option `--search`. For example:
...
-
-Here is what happens. `krun` can only explore as much of the transition
-system associated to a program as `kompile` allowed the generated language
-model to yield. Since most of the K users are interested in language models
-that execute efficiently, that is, in faster interpreters for the defined
-languages, by default `kompile` optimizes the generated language model for
-execution. In particular, it inserts no backtracking markers, which `krun`
-uses when called with the `--search` option in order to systematically generate
-the entire transition system associated to a program. This is why `krun`
-showed us only one solution when run with the `--search` option on `div.imp`.
```
Comment (Scott-Guest) [inline k-distribution/pl-tutorial/1_k/4_imp++/lesson_3/README.md:44]: Does `--search` still only return one solution here, or does it actually show all the different interleavings nowadays?

If it only returns one solution, I think we still need to include a good chunk of the prose here to explain why the other solutions are missing.

**Review by Scott-Guest — COMMENTED** [review]
Mostly LGTM, but see my comments.

One thing that's still unclear to me - the outdated lessons state that `--search` will not explore all heating / cooling interleavings unless specifically instructed to do so with `--transition`. Is that still the case, and is there some modern alternative to `--transition`? Or does `--search` explore all interleavings by default nowadays?

File: k-distribution/pl-tutorial/1_k/4_imp++/lesson_3/README.md (hunk 3 above)
Comment (gtrepta) [reply k-distribution/pl-tutorial/1_k/4_imp++/lesson_3/README.md:44]: I tested it and indeed `--search` returns all solutions now. I believe this was written before the search feature in the llvm backend was a thing.

**Review by Baltoli — APPROVED** [review]
This looks good; happy to merge with Scott's comments addressed given that there is ongoing work to slice out and review the PL tutorial anyway so any snags there aren't critical.

**Comment by gtrepta** [comment]
Closing as it's overridden by #3901 . Will open this in the pl-tutorial repo.

**Pull request closed without merging**